### PR TITLE
#495: Implementation `JSON.FORGET` command

### PR DESCRIPTION
**Pull request by Nijin-P-S**
Implementation for Issue : [495](https://github.com/DiceDB/dice/issues/494)

- Add JSON.FORGET command implementation , refer : [JSON.FORGET](https://redis.io/docs/latest/commands/json.forget/)
- Add Unit Test for JSON.FORGET 
- Add Integration Test for JSON.FORGET

Attaching below the screenshots of behavior of:
**1. JSON.FORGET command in Redis:**

Screenshot 2024-09-10 at 9 59 44 PM

**2. JSON.FORGET command in dice**
Screenshot 2024-09-10 at 9 59 50 PM

### CODE REVIEW

**Comment by kushal0511-not** [comment]
Thanks for contributing @Nijin-P-S .
I see JSON.FORGET is just alias of JSON.DEL command in docs. So instead of writing same code again can we just call JSON.DEL command inside it.wdyt?
Correct if me i am missing something.

**Comment by Nijin-P-S** [comment]
Yes @kushal0511-not ,
That should work, as JSON.FORGET is just used for deletion of a single path and JSON.DEL can be used for bulk deletion.
Calling the DEL function inside the JSON.FORGET would be better. Let me make the changes necessary and remove the excessive code.

**Comment by Nijin-P-S** [comment]
@kushal0511-not ,
Have updated the implementation to call the evalJSONDEL() thus reducing the code repeatation. 
Cross verified the unit tests, Integration tests and functionality, and is working as expected.

Attaching the screenshots:
1. Unit Tests
Screenshot 2024-09-10 at 10 41 22 PM

2. Integration Test
Screenshot 2024-09-10 at 10 41 48 PM

3. Functionality
Screenshot 2024-09-10 at 10 44 51 PM


Let me know if anything needs to be modified or added.

Thanks.

File: internal/eval/eval.go [hunk 1]
```diff
@@ -323,6 +323,30 @@ func evalGETDEL(args []string, store *dstore.Store) []byte {
...
+// If the JSONPath points to the root of the JSON document, the entire key is deleted from the store.
+// Returns an integer reply specified as the number of paths deleted (0 or more)
+func evalJSONFORGET(args []string, store *dstore.Store) []byte {
+	if len(args) < 1 {
+        return diceerrors.NewErrArity("JSON.FORGET")
+    }
+    
+    key := args[0]
+
+	// Default path is root if not specified
```
Comment (kushal0511-not) [inline internal/eval/eval.go:339]: nit: Consider aligning the comment with the code for better readability.

**Comment by Nijin-P-S** [comment]
@kushal0511-not ,
Made the changes.

Thanks.

File: internal/eval/eval.go [hunk 2]
```diff
@@ -327,6 +327,30 @@ func evalGETDEL(args []string, store *dstore.Store) []byte {
...
+// Returns response.RespZero if key is expired or it does not exist
+// Returns encoded error response if incorrect number of arguments
+// If the JSONPath points to the root of the JSON document, the entire key is deleted from the store.
+// Returns an integer reply specified as the number of paths deleted (0 or more)
+func evalJSONFORGET(args []string, store *dstore.Store) []byte {
+    if len(args) < 1 {
+        return diceerrors.NewErrArity("JSON.FORGET")
+    }
+    
+    key := args[0]
```
Comment (lucifercr07) [inline internal/eval/eval.go:341]: Can we just change L341 to L349 to return `evalJSONDEL(args, store)`?

**Review by lucifercr07 — CHANGES_REQUESTED** [review]
@Nijin-P-S thanks for contributing, have minor comment. Please review once.

**Pull request merged**